### PR TITLE
Adapt document classes

### LINKED ISSUE #1176: Changes for class Document(Base)
In the new specifications the classes DocumentBase, Document and Article are fusioned and the inheritance is dropped. 
This implies modifications on different levels:
### Standard model definitions for db:
https://github.com/openoereb/pyramid_oereb/tree/master/pyramid_oereb/standard/models
- [x] class Article to be removed (backward compatibility?)
- [x] class DocumentBase fusioned with class Document
- [x] class Document restructured
   - [x] official_titel - removed
   - [x] official_number - newly multilingual
   - [x] canton - removed (should we keep it?)
   - [x] municipality - removed for extract (usefull to keep it?)
   - [x] only_in_municipality - kind of replaces municipality for the transfer structure
   - [x] text_at_web - moved here from DocumentBase
   - [x] law_status - inherited from DocumentBase
   - [x] article_nb - new attribute for concerned articles
   - [x] index - new, integer to sort the document list
   - [x] published_until - new expiry date for temporary documents

### PLR sources definition:
https://github.com/openoereb/pyramid_oereb/blob/master/pyramid_oereb/standard/sources/plr.py

### Record definition:
https://github.com/openoereb/pyramid_oereb/blob/master/pyramid_oereb/contrib/sources/document.py

### Import Script:
https://github.com/openoereb/pyramid_oereb/blob/master/pyramid_oereb/standard/xtf_import/document.py

### Renderer:
https://github.com/openoereb/pyramid_oereb/tree/master/pyramid_oereb/lib/renderer/extract/templates/xml

### Oereblex Adapter?
https://github.com/openoereb/pyramid_oereb/blob/master/pyramid_oereb/contrib/sources/document.py

### PR DESCRIPTION
**Pull request by kdeininger**
Closes #1176 and #1249

### CODE REVIEW

**Review by vvmruder — CHANGES_REQUESTED** [review]
There are 18 tests failing for now when I do a `make clean test`. Especially the tests about `FAILED tests/renderer/test_json.py::test_format_` should be fixed.

```
=========================================== short test summary info ============================================
FAILED tests/renderer/test_json.py::test_format_real_estate - TypeError: 'NoneType' object is not subscriptable
FAILED tests/renderer/test_json.py::test_format_plr[parameter0] - TypeError: 'NoneType' object is not subscri...
FAILED tests/renderer/test_json.py::test_format_plr[parameter1] - TypeError: 'NoneType' object is not subscri...
FAILED tests/renderer/test_json.py::test_format_document[document0-result_dict0] - TypeError: 'NoneType' obje...
FAILED tests/renderer/test_json.py::test_format_document[document1-result_dict1] - TypeError: 'NoneType' obje...
FAILED tests/renderer/test_json.py::test_format_document[document2-result_dict2] - TypeError: 'NoneType' obje...
FAILED tests/renderer/test_xml.py::test_extract_against_schema[parameter0-test_extract0] - lxml.etree.Documen...
FAILED tests/renderer/test_xml.py::test_extract_against_schema[parameter1-test_extract1] - lxml.etree.Documen...
FAILED tests/renderer/test_xml.py::test_extract_against_schema[parameter2-test_extract2] - lxml.etree.Documen...
FAILED tests/webservice/test_case_insensitivity.py::test_get_extract_by_id[params0] - TypeError: 'NoneType' o...
FAILED tests/webservice/test_case_insensitivity.py::test_get_extract_by_id[params1] - TypeError: 'NoneType' o...
FAILED tests/webservice/test_case_insensitivity.py::test_get_extract_by_id[params2] - TypeError: 'NoneType' o...
FAILED tests/webservice/test_case_insensitivity.py::test_get_extract_by_id[params3] - TypeError: 'NoneType' o...
FAILED tests/webservice/test_case_insensitivity.py::test_get_extract_by_id[params4] - TypeError: 'NoneType' o...
FAILED tests/webservice/test_case_insensitivity.py::test_get_extract_by_id[params5] - TypeError: 'NoneType' o...
FAILED tests/webservice/test_getextractbyid.py::test_return_json[ALL] - TypeError: 'NoneType' object is not s...
FAILED tests/webservice/test_getextractbyid.py::test_return_json[ALL_FEDERAL] - TypeError: 'NoneType' object ...
FAILED tests/xtf_import/test_federal_topic.py::test_load - SystemExit: 1
============================ 18 failed, 356 passed, 1 warning in 181.94s (0:03:01) =============================
make: *** [Makefile:151: test] Error 1
```

Otherwise it looks good.

**Comment by kdeininger** [comment]
I have rebased this branch onto master and fixed some tests for the `OEREBlexSource` that occurred after switching the GEO-Link schema to version 1.2.1. Now I have again only the four known tests failing.

@vvmruder, did you rebuild your local devcontainers and run a `make clean-all` before running the checks?

**Comment by vuilleumierc** [comment]
I also have only 4 failing tests.
But there are some issues with `make serve-dev`:
1. I have to run it twice for it to complete, on the first try I get a Python error from load_sample_data.py
2. The service does not seem to work, the URL `http://localhost:6543/oereb/extract/reduced/xml/CH113928077734` gives an empty content (same with JSON). But maybe this is normal at this point?

**Comment by vvmruder** [comment]
I tried it now and the tests are ok.

But now I get an error while dev date is filled into the db:
```
.venv/bin/python pyramid_oereb/standard/load_sample_data.py --configuration pyramid_oereb/standard/pyramid_oereb.yml --sql-file .db/13-fill.sql --dir sample_data
Import theme contaminated_public_transport_sites.
Traceback (most recent call last):
  File "pyramid_oereb/standard/load_sample_data.py", line 329, in <module>
    _run()
  File "pyramid_oereb/standard/load_sample_data.py", line 324, in _run
    sql_file=sql_file
  File "pyramid_oereb/standard/load_sample_data.py", line 236, in load
    ][0]
IndexError: list index out of range
make: *** [Makefile:106: .db/13-fill.sql] Error 1
```

for `make serve-dev`

**Comment by kdeininger** [comment]
Okay, I will fix the sample data.

**Comment by kdeininger** [comment]
The sample data import should work now.

**Comment by vuilleumierc** [comment]
Looks good to me, thanks!
I have one comment in the code

File: pyramid_oereb/lib/records/documents.py [hunk 1]
```diff
@@ -6,317 +6,115 @@
...
+        if official_number and not isinstance(official_number, dict):
+            warnings.warn('Type of "official_number" should be "dict"')
+        if published_from and not isinstance(published_from, datetime.date):
+            warnings.warn('Type of "published_from" should be "datetime.date", not '
+                          + str(type(published_from)))
+        if text_at_web and not isinstance(text_at_web, dict):
+            warnings.warn('Type of "text_at_web" should be "dict"')
         if published_from and not isinstance(published_from, datetime.date):
             warnings.warn('Type of "published_from" should be "datetime.date", not '
                           + str(type(published_from)))
```
Comment (vuilleumierc) [inline pyramid_oereb/lib/records/documents.py:71]: shouldn't we test here for the type of the new attributes `only_in_municipality`, `article_nb` and `published_until`?

Comment (vvmruder) [reply pyramid_oereb/lib/records/documents.py:71]: This might be a good completion :D

**Comment by vvmruder** [comment]
Maybe solve the comment above. Otherwise Iam happy with it and thank you for this bunch of work.

File: pyramid_oereb/lib/records/documents.py (hunk 1 above)
Comment (kdeininger) [reply pyramid_oereb/lib/records/documents.py:90]: Done 😉

**Review by vuilleumierc — APPROVED** [review]

**Review by vvmruder — APPROVED** [review]

**Pull request merged**